curriculum/challenges/english/07-scientific-computing-with-python/learn-tree-traversal-by-building-a-binary-search-tree/65c9ddd336596e30a4266a50.md

```diff
@@ -0,0 +1,93 @@
+---
+id: 65c9ddd336596e30a4266a50
+title: Step 31
+challengeType: 20
+dashedName: step-31
+---
+
+# --description--
+
+In the body of the `__str__` method, delete `pass` and return the result of calling the `str()` function with `self.key` as the argument.  This is the attribute of the current node object that stores the value associated with the node.
```

Comment:
Reminder: Instruction should be rephrased as the `__str__` method is already discussed in detail in the refactored sudoku solver.